crates/ruff_python_formatter/src/expression/parentheses.rs

```diff
@@ -173,30 +170,10 @@ impl<'ast> Format<PyFormatContext<'ast>> for FormatParenthesized<'_, 'ast> {
         let indented = format_with(|f| {
             let content = Arguments::from(&self.content);
             if self.comments.is_empty() {
-                match self.hug {
-                    None => group(&soft_block_indent(&content)).fmt(f),
-                    Some(HuggingStyle::Always) => content.fmt(f),
-                    Some(HuggingStyle::IfFirstLineFits) => {
-                        // It's not immediately obvious how the below IR works to only indent the content if the first line exceeds the configured line width.
-                        // The trick is the first group that doesn't wrap `self.content`.
-                        // * The group doesn't wrap `self.content` because we need to assume that `self.content`
-                        //   contains a hard line break and hard-line-breaks always expand the enclosing group.
-                        // * The printer decides that a group fits if its content (in this case a `soft_line_break` that has a width of 0 and is guaranteed to fit)
-                        //   and the content coming after the group in expanded mode (`self.content`) fits on the line.
-                        //   The content coming after fits if the content up to the first soft or hard line break (or the end of the document) fits.
-                        //
-                        // This happens to be right what we want. The first group should add an indent and a soft line break if the content of `self.content`
-                        // up to the first line break exceeds the configured line length, but not otherwise.
-                        let indented = f.group_id("indented_content");
-                        write!(
-                            f,
-                            [
-                                group(&indent(&soft_line_break())).with_group_id(Some(indented)),
-                                indent_if_group_breaks(&content, indented),
-                                if_group_breaks(&soft_line_break()).with_group_id(Some(indented))
-                            ]
-                        )
-                    }
```

Comment:
So this got much simpler because we no longer check if indenting would allow us to avoid line-length violations?

Reply:
Yes. Breaking only if the first line doesn't fit required a more complicated IR. Now it's simpler, because we always or never indent.